src/ddmd/dclass.d

```diff
@@ -212,10 +232,10 @@ extern (C++) class ClassDeclaration : AggregateDeclaration
     // the ClassInfo object for this ClassDeclaration
     TypeInfoClassDeclaration vclassinfo;
 
-    bool com;           // true if this is a COM class (meaning it derives from IUnknown)
-    bool cpp;           // true if this is a C++ interface
-    bool isobjc;        // true if this is an Objective-C class/interface
-    bool isscope;       // true if this is a scope class
+    /* bitfield that keeps track if the class is com, cpp, objc, scope or anonymous
+     */
+    ubyte classProperties;
```

Comment:
I recommend using Ddoc comment for this as well. Keep in mind that the compiler is exposed as a library. In my opinion all public/protected symbols should have Ddoc documentation. I don't expect that the documentation will be added to all symbols at once, but we should avoid making it worse 😃.

Reply:
`classProperties` has no obvious relationship with the `ClassFlagMask` values. Call the fields `classKind` and `classFlags`, to match up with `ClassKind` values and `ClassFlags` flag bits.

Reply:
ClassFlags is already taken and is a struct [1], but it's not used in dclass at all; toobj.d is the only file that uses it, so I guess we can move it there and use ClassFlags in dclass. I thought about modifying the ClassFlags struct so that it accommodates the anonymous, stack and objc flags as well, but it seems, if I am not mistaking, that the backend expects a certain layout of that class.

[1] https://github.com/dlang/dmd/blob/master/src/ddmd/dclass.d#L154

Reply:
If `ClassFlags` is already taken, what about `ClassKind`?